Apps/WebClient/src/ClientApp/app/components/timeline/comment.vue

```diff
@@ -4,28 +4,253 @@
+                  size="1x"
+                ></font-awesome-icon>
+              </template>
+              <b-dropdown-item class="menuItem" @click="editComment()">
+                Edit
+              </b-dropdown-item>
+              <b-dropdown-item class="menuItem" @click="deleteComment()">
+                Delete
+              </b-dropdown-item>
+            </b-nav-item-dropdown>
+          </b-navbar-nav>
+        </div>
+      </b-col>
     </b-row>
-    <b-row class="px-3">
+    <b-row
+      v-if="mode !== 'text'"
+      class="comment-body py-2 my-1"
+      align-v="center"
+    >
+      <div v-if="mode === 'add'">
+        <div
+          :id="'tooltip-' + comment.parentEntryId"
+          class="tooltip-info d-flex pl-2"
+        >
+          <font-awesome-icon :icon="lockIcon" size="1x"> </font-awesome-icon>
+        </div>
+        <b-tooltip
+          variant="secondary"
+          :target="'tooltip-' + comment.parentEntryId"
+          placement="left"
+          triggers="hover"
+        >
+          Only you can see comments added to your medical records.
+        </b-tooltip>
+      </div>
+      <div class="comment-input pl-2">
+        <b-form @submit.prevent>
+          <b-form-textarea
+            rows="1"
+            no-resize
+            ref="commentInput"
+            v-model="commentInput"
+            :placeholder="placeholder"
+            maxlength="1000"
+          ></b-form-textarea>
+        </b-form>
+      </div>
+      <div class="d-flex comment-button px-3 flex-row">
+        <b-button
+          variant="primary"
+          @click="onSubmit"
+          :disabled="commentInput === ''"
+          class="d-flex"
+        >
+          Save
+        </b-button>
+        <div v-if="mode === 'edit'" class="d-flex pl-2">
+          <b-button variant="secondary" @click="onCancel">
+            Cancel
+          </b-button>
+        </div>
+      </div>
+    </b-row>
+    <b-row class="px-3" v-if="mode !== 'add'">
       <span> {{ formatDate(comment.createdDateTime) }} </span>
     </b-row>
   </b-col>
 </template>
 <script lang="ts">
 import Vue from "vue";
 import UserComment from "@/models/userComment";
-import { Prop, Component } from "vue-property-decorator";
+import User from "@/models/user";
+import { Getter } from "vuex-class";
+import { Prop, Component, Emit, Watch } from "vue-property-decorator";
+import {
+  faEllipsisV,
+  IconDefinition,
+  faLock,
+} from "@fortawesome/free-solid-svg-icons";
+import { IUserCommentService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
 
 @Component
 export default class CommentComponent extends Vue {
-  @Prop({ default: "Default" }) comment!: UserComment;
+  @Getter("user", { namespace: "user" }) user!: User;
+  @Prop() comment!: UserComment;
+  private commentInput: string = "";
+  private commentService!: IUserCommentService;
+  private mode: string = "text";
```

Comment:
would it be possible to infer if its edit or add based on the comment id?

Reply:
in mounted it checks if the id === "", if it is the mode is set to add. id doesn't change on edit